metals/src/main/scala/scala/meta/internal/metals/SemanticdbIndexer.scala

```diff
@@ -67,7 +67,7 @@ class SemanticdbIndexer(
       try {
         stream.forEach {
           case path if path.isSemanticdb =>
-            SemanticdbPath(AbsolutePath(path))
+            onChange(SemanticdbPath(AbsolutePath(path)))
```

Comment:
Soooo, this is really bad :| and probably my fault.

Everything else is about the test. If anyone feels the test is not ok, I would at least merge the change.

Reply:
I think it's fine. If it's actually testing what we want, then I'd just leave the hackiness in there.